pkg/config/agentmanagement.go

```diff
@@ -37,22 +40,67 @@ func newRemoteConfigHTTPProvider(c *Config) (*remoteConfigHTTPProvider, error) {
 	}, nil
 }
 
+type remoteConfigCache struct {
+	UrlHash string `json:"url_hash"`
+	Config  string `json:"config"`
+}
+
+func hashUrl(u string) string {
```

Comment:
would we want to invalidate the cache when other fields change? (for example the namespace or labelset)

Reply:
For `namespace`, at least, I'd imagine so, as it could have an entirely different base config. For `labelset`, I'd personally think also yes. 

For example, if one were to stop a service running on a node (say previously a database was running but now is not) and remove the associated label, if the agent were restarted with the new `labelset` and unable to fetch the `remote_config` it would use the old cached config that had the snippets related to the now-stopped database process.